src/test/addrman_tests.cpp

```diff
@@ -47,11 +47,12 @@ static CService ResolveService(const std::string& ip, uint16_t port = 0)
 }
 
 
-static std::vector<bool> FromBytes(const unsigned char* source, int vector_size)
+static std::vector<bool> FromBytes(std::span<const std::byte> source)
 {
+    int vector_size(source.size() * 8);
     std::vector<bool> result(vector_size);
     for (int byte_i = 0; byte_i < vector_size / 8; ++byte_i) {
-        unsigned char cur_byte = source[byte_i];
+        uint8_t cur_byte{uint8_t(source[byte_i])};
```

Comment:
Would it better to use `std::to_integer` here? It does the same thing, but maybe it would be good to use it for its compile-time guarantees.